src/backend/workers/go.remoteIDs/identities.go

```diff
@@ -0,0 +1,168 @@
+		}
+		db, err := store.InitializeCassandraBackend(c)
+		if err != nil {
+			log.WithError(err).Warnf("[initDbHandler] initialization of %s backend failed", poller.Config.StoreName)
+			return handler, errors.New("[initDbHandler] failed to init cassandra backend")
+		}
+		handler.Store = db
+		handler.cache = make(map[string]cacheEntry)
+		handler.cacheMux = new(sync.Mutex)
+		return handler, nil
+	default:
+		return handler, fmt.Errorf("[initDbHandler] unhandled store : %s", poller.Config.StoreName)
+	}
+}
+
+// SyncCache resync DbHandler's memory cache with 'active' remote identities found in db
+func (dbh *DbHandler) SyncCache() (added, removed, updated map[string]cacheEntry, err error) {
+	added = make(map[string]cacheEntry)
+	removed = make(map[string]cacheEntry)
+	updated = make(map[string]cacheEntry)
+	active := make(map[string]cacheEntry)
+	const defaultInterval = "15"
+
+	remotes, err := dbh.Store.RetrieveAllRemotes(false)
+	if err != nil {
+		log.WithError(err).Warn("[updateCache] failed to retrieve remote identities")
+		err = errors.New("[updateCache] failed to retrieve remote identities")
+		return
+	}
+
+	dbh.cacheMux.Lock()
+	defer dbh.cacheMux.Unlock()
+	for remote := range remotes {
+		if dbh.statusTypeOK(remote) {
+			idkey := remote.UserId.String() + remote.Id.String()
+			active[idkey] = cacheEntry{}
+			if entry, ok := dbh.cache[idkey]; ok {
+				//check if pollinterval has changed
+				var pollInterval string
+				var ok bool
+				if pollInterval, ok = remote.Infos["pollinterval"]; !ok || pollInterval == "" {
+					// do not resign, take a default value instead
+					pollInterval = defaultInterval
+				}
+				if entry.pollInterval != pollInterval {
+					entry.pollInterval = pollInterval
+					updated[idkey] = entry
+					dbh.cache[idkey] = entry
+				}
+			} else {
+				var pollInterval string
+				var ok bool
+				if pollInterval, ok = remote.Infos["pollinterval"]; !ok || pollInterval == "" {
+					// do not resign, take a default value instead
+					pollInterval = defaultInterval
+				}
+				entry := cacheEntry{
+					iDkey:          idkey,
+					pollInterval:   pollInterval,
+					remoteID:       remote.Id,
+					remoteProtocol: remote.Protocol,
+					userID:         remote.UserId,
+				}
+				dbh.cache[idkey] = entry
+				added[idkey] = entry
+			}
+		}
+	}
+	for key, entry := range dbh.cache {
+		if _, ok := active[key]; !ok {
+			removed[key] = entry
+			delete(dbh.cache, key)
+		}
+	}
+	return
+}
+
+// statusTypeOK checks if remote identity is 'active' and its type is within poller's scope
+func (dbh *DbHandler) statusTypeOK(remote *UserIdentity) bool {
+	if remote.Status != "active" {
+		return false
+	}
+	for _, t := range dbh.remoteProtocols {
+		if t == remote.Protocol {
+			return true
+		}
+	}
+	return false
+}
+
+func (dbh *DbHandler) GetCacheEntry(key string) (cacheEntry, bool) {
+	dbh.cacheMux.Lock()
+	entry, ok := dbh.cache[key]
+	dbh.cacheMux.Unlock()
```

Comment:
unlock using defer

Reply:
useless because it cannot fails between Lock and Unlock